src/app/clusters/water-heater-management-server/water-heater-management-server.h

```diff
@@ -0,0 +1,133 @@
+     *                  to the previous mode (e.g. OFF, MANUAL or TIMED).
+     * @param oneShot  Indicates whether the BOOST state should be automatically canceled once the hot water has first reached the
+     *                 set point temperature (or the TemporarySetpoint temperature, if specified) for the TargetPercentage (if specified).
+     * @param emergencyBoost  Indicates that the consumer wants the water to be heated as quickly as practicable. This MAY cause
+     *                        multiple heat sources to be activated (e.g. a heat pump and direct electric heating element).
+     * @param temporarySetpoint  Indicates the target temperature to which to heat the hot water for this Boost command. It SHALL be
+     *                           used instead of the normal set point temperature whilst the BOOST state is active.
+     * @param targetPercentage  If the tank supports the TankPercent feature, this field indicates the amount of water that SHALL be
+     *                          heated by this Boost command before the heater is switched off.
+     * @param targetReheat  If the tank supports the TankPercent feature, and the heating by this Boost command has ceased because
+     *                      the TargetPercentage of the water in the tank has been heated to the set point (or TemporarySetpoint if included),
+     *                      this field indicates the percentage to which the hot water in the tank SHALL be allowed to fall before
+     *                      again beginning to reheat it.
+     *
+     * @return  Success if the boost command is accepted; otherwise the command SHALL be rejected with appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status HandleBoost(uint32_t duration, Optional<bool> oneShot,
+                                                            Optional<bool> emergencyBoost, Optional<int16_t> temporarySetpoint,
+                                                            Optional<chip::Percent> targetPercentage,
+                                                            Optional<chip::Percent> targetReheat) = 0;
+
+    /**
+     * @brief Delegate should implement a handler to cancel a boost command.
+     *        Upon receipt, the Water Heater SHALL transition back from the BOOST state to the previous mode (e.g. OFF,
+     *        MANUAL or TIMED).
+     *
+     * @return It should report SUCCESS if successful and FAILURE otherwise.
+     */
+    virtual Protocols::InteractionModel::Status HandleCancelBoost() = 0;
+
+    // ------------------------------------------------------------------
+    // Get attribute methods
+    virtual BitMask<WaterHeaterTypeBitmap> GetHeaterTypes()  = 0;
+    virtual BitMask<WaterHeaterDemandBitmap> GetHeatDemand() = 0;
+    virtual uint16_t GetTankVolume()                         = 0;
+    virtual int64_t GetEstimatedHeatRequired()               = 0;
```

Comment:
Why are we forcing a delegate that does not implement the EM feature to still implement these methods?

Similar for GetTankPercentage and the TP feature.

Reply:
It is a reasonable point. Say we provide an implementation in this header file which returns 0, should we do anything about checking whether the EM feature is supported and if not I guess a do a minimum of printing an error message but we could get more vicious as this would be a programming bug?

Reply:
I don't have strong opinions here.  I'm fine with us just forcing all delegates to implement these too, as long as we're doing that purposefully.